hack/make-rules/test-cmd-util.sh

```diff
@@ -2932,7 +2932,7 @@ runTests() {
 
   # Make sure the UI can be proxied
   start-proxy
-  check-curl-proxy-code /ui 301
+  check-curl-proxy-code /ui 307
```

Comment:
The "normal" handler appears to handle a registration of `/ui/` by creating a 301 from `/ui` to `/ui/`.  We serve it directly out of `/ui` which means the handler's actual `307` comes back instead of a 301 to a 307.